src/components/forms/validator.ts

```diff
@@ -27,104 +21,84 @@ export const required = (value?: string) => {
   return undefined
 }
 
-export const mustBeInteger = (value: string) =>
+export const mustBeInteger = (value: string): ValidatorReturnType =>
   !Number.isInteger(Number(value)) || value.includes('.') ? 'Must be an integer' : undefined
 
-export const mustBeFloat = (value: string) => (value && Number.isNaN(Number(value)) ? 'Must be a number' : undefined)
-
-export const greaterThan = (min: number | string) => (value: string) => {
-  if (Number.isNaN(Number(value)) || Number.parseFloat(value) > Number(min)) {
-    return undefined
-  }
-
-  return `Should be greater than ${min}`
-}
-
-export const equalOrGreaterThan = (min: number | string) => (value: string): undefined | string => {
-  if (Number.isNaN(Number(value)) || Number.parseFloat(value) >= Number(min)) {
-    return undefined
-  }
-
-  return `Should be equal or greater than ${min}`
-}
+export const mustBeFloat = (value: string): ValidatorReturnType =>
+  value && Number.isNaN(Number(value)) ? 'Must be a number' : undefined
 
 const regexQuery = /^(?:(?:https?|ftp):\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,}))\.?)(?::\d{2,5})?(?:[/?#]\S*)?$/i
 const url = new RegExp(regexQuery)
-export const mustBeUrl = (value: string) => {
+export const mustBeUrl = (value: string): ValidatorReturnType => {
   if (url.test(value)) {
     return undefined
   }
 
   return 'Please, provide a valid url'
 }
 
-export const minValue = (min: number | string) => (value: string) => {
-  if (Number.isNaN(Number(value)) || Number.parseFloat(value) >= Number(min)) {
+export const minValue = (min: number | string, inclusive = true) => (value: string): ValidatorReturnType => {
+  if (Number.parseFloat(value) > Number(min) || (inclusive && Number.parseFloat(value) >= Number(min))) {
     return undefined
   }
 
   return `Should be at least ${min}`
```

Comment:
If we use the `inclusive = false`, this message looks odd.

Also, why are we checking that there's a `toBeDefined` return value in the tests and not the expected message?

I'm just thinking, if what we're checking is that the validation is correct, then we can consider abstracting the validation itself to a function that returns a boolean value and wrap it into the validation function that will return `undefined` or an expected error message.

```typescript
const isSmallerThan = (min: number | string, value: number | string, inclusive = true): boolean => {
  return (Number.parseFloat(value.toString()) > Number(min) || (inclusive && Number.parseFloat(value.toString()) >= Number(min)))
}

const minValue = (min: number | string, inclusive = true) => (value: string): ValidatorReturnType => {
  if (isSmallerThan(min, value, inclusive)) {
    return
  }
  
  if (inclusive) {
    return `Should be at least ${min}`
  } else {
    return `Should be greater than ${min}`
  }
}

const maxValue = (max: number | string) => (value: string) ValidatorReturnType => {
  if (isSmallerThan(value, max, true)) {
    return
  }

  return `Should be at most ${max}`
}
```

Then you can test the validator itself and the `isSmallerThan` function apart (BTW, I'm not so sure of the function name, I just want to make my point clear)

And, we can consider using `bignumber.js` or whatever BN library we're going to use to make those numbers comparisons as we're working with more than 16 decimal digits.

WDY?

Reply:
TBH I only added tests because coveralls CI job wouldn't let me merge this PR 😄

Reply:
> we can consider abstracting the validation itself to a function that returns a boolean value

In our app the validators are unique to forms, so the expected return value is either undefined or an error message, thus. I wouldn't spend time on it

Reply:
Yeah checking the messages makes sense, will fix it

Reply:
> And, we can consider using bignumber.js or whatever BN library we're going to use to make those numbers comparisons as we're working with more than 16 decimal digits.

We don't have a form where one can put a wei number, do we?

Reply:
well... send funds allows us to send any value for a token (or Eth), so yes, we can put a wei number in there.

Same for the ContractInteraction, where you can put the value you want to send to the contract.

Reply:
also, we can consider using https://github.com/protofire/big-number-input for those bn inputs. It's battle-tested and Nico already used it in the safe apps.

Reply:
> well... send funds allows us to send any value for a token (or Eth), so yes, we can put a wei number in there.

Doesn't seem so. At least on the review screen we convert input value to wei value: https://github.com/gnosis/safe-react/blob/development/src/routes/safe/components/Balances/SendModal/screens/ReviewTx/index.tsx#L63

So the validated amount is in ETH

Reply:
This is the actual LOC for conversion: https://github.com/gnosis/safe-react/blob/development/src/routes/safe/components/Balances/SendModal/screens/ReviewTx/index.tsx#L90